pallets/time-release/src/lib.rs

```diff
@@ -324,24 +349,24 @@ pub mod module {
 		) -> DispatchResult {
 			ensure_signed(origin)?;
 			let who = T::Lookup::lookup(dest)?;
-			let locked_amount = Self::do_claim(&who);
+			let frozen_amount = Self::do_claim(&who)?;
 
-			Self::deposit_event(Event::Claimed { who, amount: locked_amount });
+			Self::deposit_event(Event::Claimed { who, amount: frozen_amount });
 			Ok(())
 		}
 	}
 }
 
 impl<T: Config> Pallet<T> {
-	fn do_claim(who: &T::AccountId) -> BalanceOf<T> {
-		let locked = Self::prune_and_get_locked_balance(who);
-		if locked.is_zero() {
-			Self::delete_lock(who);
+	fn do_claim(who: &T::AccountId) -> Result<BalanceOf<T>, DispatchError> {
+		let frozen = Self::prune_and_get_frozen_balance(who);
+		if frozen.is_zero() {
+			Self::delete_freeze(who)?;
 		} else {
-			Self::update_lock(who, locked);
+			Self::update_freeze(who, frozen)?;
```

Comment:
Not worth the change but good to know that if we set the update_freeze frozen amount to zero it is the same thing as calling thaw. I remove the if statement.